src/main/java/io/curity/identityserver/plugin/OptInMFA/AuthenticatorModel.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright 2020 Curity AB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.curity.identityserver.plugin.OptInMFA;
+
+import se.curity.identityserver.sdk.Nullable;
+import se.curity.identityserver.sdk.service.authenticationaction.AuthenticatorDescriptor;
+
+public final class AuthenticatorModel
+{
+    private final String _acr;
+    private final String _description;
+    private final String _type;
+
+    public AuthenticatorModel(String acr, @Nullable String description, String type, String name)
+    {
+        _acr = acr;
+
+        if (!name.equals(acr)) {
+            _description = name;
+        }
+        else if (description != null)
+        {
+            _description = description;
+        }
+        else
+        {
+            _description = acr;
+        }
+
+        _type = type;
+    }
+
+    public String getAcr()
+    {
+        return _acr;
+    }
+
+    public String getDescription()
+    {
+        return _description;
+    }
+
+    public String getType()
+    {
+        return _type;
+    }
+
+    public static AuthenticatorModel toAuthenticatorModel(AuthenticatorDescriptor descriptor, String name)
+    {
+        //TODO - there should be a way to get the type from somewhere
+        return new AuthenticatorModel(descriptor.getAcr(), descriptor.getDescription(), "html-form", name);
```

Comment:
The type normally comes from `PluginDescriptor#getImplementationType`.

Reply:
I need the type of the authenticator, not the plugin. This type is needed to show proper icon on the list of authenticators.

Reply:
I think you mean the plugin `ID`, not type, in this case?

Reply:
No, it's the type of the authenticator, not the plugin. The `type` property of an authenticator is unfortunately not available in the SDK. This value is used to choose the proper icon when displaying list of authenticators. So that when a user has "google" authenticator on their list of available second factors, we can properly show the "G" icon next to this authenticator option.